addons/web/static/src/views/fields/selection/selection_field.js

```diff
@@ -49,7 +49,7 @@ export class SelectionField extends Component {
                     : "";
             case "selection":
                 return this.props.record.data[this.props.name] !== false
-                    ? this.options.find((o) => o[0] === this.props.record.data[this.props.name])[1]
+                    ? this.options.find((o) => o[0] === this.props.record.data[this.props.name])?.[1]
```

Comment:
how could the field be set to an option that doesn't exist? This doesn't make sense.

Reply:
In the new dynamic selection field, we can have no option simply just because there's no valid options available. The `this.options` is now dynamic

Reply:
you should override `get string()` is the dynamic selection field then. This doesn't make sense here and could hide real issues.